iceaxe/__tests__/test_text_search.py

```diff
@@ -0,0 +1,174 @@
+from typing import List, Literal, Optional, TypeVar, cast
+
+import pytest
+
+from iceaxe import Field, TableBase, func, select
+from iceaxe.postgres import PostgresFullText
+from iceaxe.queries import QueryBuilder
+from iceaxe.session import DBConnection
+
+T = TypeVar("T")
+
+
+class Article(TableBase):
+    """Test model for full-text search."""
+
+    id: int = Field(primary_key=True)
+    title: str = Field(postgres_config=PostgresFullText(language="english", weight="A"))
+    content: str = Field(
+        postgres_config=PostgresFullText(language="english", weight="B")
+    )
+    summary: Optional[str] = Field(
+        default=None, postgres_config=PostgresFullText(language="english", weight="C")
+    )
+
+    async def save(self, db_connection: DBConnection) -> None:
+        """Save the article to the database."""
+        await db_connection.insert([self])
+
+
+async def execute(
```

Comment:
This casting should already be the behavior or db_connection.exec. Why reimplement the execute here?